### PR TITLE
JA: Add ja/lite/g3doc/performance/post_training_quantization.md

### PR DESCRIPTION
**Pull request by zinziroge**
Original Document:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/g3doc/performance/post_training_quantization.md

### CODE REVIEW

**Comment by zinziroge** [comment]
yes, i removed image file.

File: site/ja/lite/performance/post_training_quantization.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+8ビット量子化は浮動小数点数を以下の式で近似します。
+
+$$real\_value = (int8\_value - zero\_point) \times scale$$
+
+その表現はおおきく2つの部分があります。
+
+* ゼロ点が0に等しく、-127以上127以下の8ビットの2の補数で表現された軸毎(つまりチャンネル毎)、あるいはテンソルごとの重み
+
+* ゼロ点が-128以上127以下のどこかにあり、-128以上127以下の8ビットの2の補数で表現されたテンソルごとの活性化と入力
```
Comment (AseiSugiyama) [inline site/ja/lite/performance/post_training_quantization.md:159]: [weak suggestion]

```suggestion
* Tensor ごとの活性化関数と入力を、-128以上127以下の 2 の補数表現を用いた int8 の整数で、ゼロ点を-128以上127以下にしたもの
```

File: site/ja/lite/performance/post_training_quantization.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# 訓練後の量子化
+
+訓練後の量子化は、わずかなモデルの精度の低下を伴いますが、
+モデルの大きさを削減することができ、さらにCPUとハードウェアアクセラレータのレイテンシーを改善する変換手法です。
+[TensorFlow Lite Converter](../convert/)
+を使って TensorFlow Lite 形式に変換するときに、訓練済みの浮動小数点数の TensorFlow モデルを使ってこれらの手法を実行できます。
+
+注意: このページの関数はTensorFlow 1.15 以上が必要です。
```
Comment (AseiSugiyama) [inline site/ja/lite/performance/post_training_quantization.md:8]: [strong suggestion]
```suggestion
Note: このページの関数はTensorFlow 1.15 以上が必要です。
```

In the following comment, I described the reason why we should leave "Note:" untranslated. (Sorry, I found this line after writing the comment describing it.)

File: site/ja/lite/performance/post_training_quantization.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import tensorflow as tf
+converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
+<b>converter.optimizations = [tf.lite.Optimize.DEFAULT]</b>
+tflite_quant_model = converter.convert()
+</pre>
+
+推論時に、重みは8ビット精度から浮動小数点数に変換され、浮動小数点数カーネルを使って計算されます。
+この変換は一度だけおこなわれ、レイテンシーを減らすためにキャッシュされます。
+
+レイテンシーをさらに改善するために、"ダイナミックレンジ" 演算子は、8ビットに合わせた範囲に基づき動的に活性化を量子化し、8ビット重みと活性化を用いて計算を実施します。
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:42]: i used your suggession.

File: site/ja/lite/performance/post_training_quantization.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+*   `tf.compat.v1.nn.bidirectional_dynamic_rnn`
+*   `tf.compat.v1.nn.dynamic_rnn`
+
+### 完全な整数量子化
+
+モデル内のすべての計算の量子化を実施することで、
+さらにレイテンシー改善したり、ピーク時のメモリ使用量を削減したり、
+整数演算のみに対応したハードウェアデバイスやアクセラレータを利用できるようになります。
+
+こうするためにはサンプル入力データを変換器に与え、活性化と入力の範囲を計測する必要があります。
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:61]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+*   `tf.keras.layers.LSTM`
+*   `tf.nn.embedding_lookup`
+*   `tf.compat.v1.nn.rnn_cell.BasicRNNCell`
+*   `tf.compat.v1.nn.bidirectional_dynamic_rnn`
+*   `tf.compat.v1.nn.dynamic_rnn`
+
+### 完全な整数量子化
+
+モデル内のすべての計算の量子化を実施することで、
+さらにレイテンシー改善したり、ピーク時のメモリ使用量を削減したり、
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:58]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+これは選択肢の概要の一覧表とその効果です。
+
+| 手法                      | 効果                      | ハードウェア            |
+| ------------------------- | ------------------------- | ------------------- |
+| ダイナミックレンジ         | 4倍小型化, 2～3倍高速化   | CPU                 |
+: 量子化                    :                          :                     :
+| 完全な整数                | 4倍小型化、3倍以上高速化   | CPU、Edge TPU       |
+: 量子化                    :                           : マイクロコントローラ :
+| 半精度浮動小数点数量子化   | 2倍小型化, GPU    | CPU, GPU             |
+:                           : アクセラレーションの可能性 :                     :
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:22]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+モデル内のすべての計算の量子化を実施することで、
+さらにレイテンシー改善したり、ピーク時のメモリ使用量を削減したり、
+整数演算のみに対応したハードウェアデバイスやアクセラレータを利用できるようになります。
+
+こうするためにはサンプル入力データを変換器に与え、活性化と入力の範囲を計測する必要があります。
+以下のコードで使用されている `representative_dataset_gen()` 関数を参照してください
+
+#### 浮動小数点数の代替をともなう整数 (元々の浮動小数点数の入力と出力を使用する)
+
+完全に整数量子化されたモデルを使うが、整数に対応した(整数への変換が円滑にできることが確かめれらた)実装がない浮動小数点演算を使用するときには、以下のステップを使ってください。
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:66]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+<b>converter.optimizations = [tf.lite.Optimize.DEFAULT]
+def representative_dataset_gen():
+  for _ in range(num_calibration_steps):
+    # あなたの選択した関数内で、サンプル入力をnumpy配列として取得する
+    yield [input]
+converter.representative_dataset = representative_dataset_gen</b>
+tflite_quant_model = converter.convert()
+</pre>
+
+注意: これは、整数のみのデバイス(8ビットマイクロコントローラのような)やアクセラレータ(Coral Edge TPUのような)と互換性がないでしょう。
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:80]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+推論時の利便性を考慮し、元の浮動小数点数のみのモデルとおなじインターフェースを保持するために、入力と出力は浮動小数点数のままです。
+
+#### 整数のみ
+
+*これは
+[マイクロコントローラ向け TensorFlow Lite](https://www.tensorflow.org/lite/microcontrollers)
+と [Coral Edge TPUs](https://coral.ai/) に対する共通のユースケースです*
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:87]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    # あなたが選択した関数内で、サンプル入力データをnumpy配列として取得する。
+    yield [input]
+converter.representative_dataset = representative_dataset_gen
+<b>converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]</b>
+<b>converter.inference_input_type = tf.int8</b>  # or tf.uint8
+<b>converter.inference_output_type = tf.int8</b>  # or tf.uint8
+tflite_quant_model = converter.convert()
+</pre>
+
+注意: 現在は量子化できない演算があった場合は、変換器はエラーを投げるでしょう。
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:107]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+
+* モデルの大きさを半分にまで削減します(すべての重みがオリジナルのサイズの半分になるので)
+* 最小限の精度劣化
+* いくつのデリゲート(たとえば、GPUへのデリゲート)は、半精度浮動小数点数データで直接演算でき、結果として単精度浮動小数点数の計算より速く実行されます。
+
+この量子化の欠点は以下のとおりです。
+
+* 最大の性能を必要とするのであれば、この量子化はよい選択ではありません (その場合は、固定小数点数への量子化の方がより良いでしょう)。
+* デフォルトでは、半精度浮動小数点数の量子化モデルは、CPUで実行するときに重みの値を単精度浮動小数点数に"逆量子化する"でしょう。
+(GPUデリゲートはこの逆量子化は行わないでしょう、その理由は半精度浮動小数点数のままで演算できるからです。)
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:132]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+
+### モデル精度
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:134]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+
+### 量子化されたテンソルの表現
+
+8ビット量子化は浮動小数点数を以下の式で近似します。
+
+$$real\_value = (int8\_value - zero\_point) \times scale$$
+
+その表現はおおきく2つの部分があります。
+
+* ゼロ点が0に等しく、-127以上127以下の8ビットの2の補数で表現された軸毎(つまりチャンネル毎)、あるいはテンソルごとの重み
```
[the remark this answers is not in the document]
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:157]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md (hunk 1 above)
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:159]: i used your suggestion.

File: site/ja/lite/performance/post_training_quantization.md (hunk 2 above)
Comment (zinziroge) [reply site/ja/lite/performance/post_training_quantization.md:8]: i used your suggestion.

**Comment by AseiSugiyama** [comment]
@zinziroge Great work, LGTM!

**Pull request merged**